src/connection/secret_connection.rs

```diff
@@ -114,68 +116,75 @@ impl<IoHandler: Read + Write + Send + Sync> SecretConnection<IoHandler> {
             }
         };
 
-        let remote_pubkey =
-            ed25519::PublicKey::from_bytes(&auth_sig_msg.key).ok_or_else(|| ErrorKind::Crypto)?;
+        let remote_pubkey = ed25519::PublicKey::from_bytes(&auth_sig_msg.key)
+            .ok_or_else(|| ErrorKind::CryptoError)?;
         let remote_signature: &[u8] = &auth_sig_msg.sig;
-        let remote_sig = ed25519::Signature::from_bytes(remote_signature)?;
+        let remote_sig =
+            ed25519::Signature::from_bytes(remote_signature).map_err(|_| ErrorKind::CryptoError)?;
 
-        let remote_verifier = Ed25519Verifier::from(&remote_pubkey);
-        remote_verifier.verify(&kdf.challenge, &remote_sig)?;
+        Ed25519Verifier::from(&remote_pubkey)
+            .verify(&kdf.challenge, &remote_sig)
+            .map_err(|_| ErrorKind::CryptoError)?;
 
         // We've authorized.
         sc.remote_pubkey = PublicKey::from(remote_pubkey);
 
         Ok(sc)
     }
 
-    /// Unseal (i.e. decrypt) AEAD authenticated data
-    fn open(&self, ciphertext: &[u8], out: &mut [u8]) -> Result<usize, Error> {
-        let nonce = aead::Nonce::from(&self.recv_nonce);
-        let associated_data = aead::Aad::empty();
-
-        // optimize if the provided buffer is sufficiently large
-        let len = if out.len() >= ciphertext.len() {
-            let in_out = &mut out[..ciphertext.len()];
-            in_out.copy_from_slice(ciphertext);
-
-            aead::open_in_place(&self.recv_secret, nonce, associated_data, 0, in_out)
-                .map_err(|_| ErrorKind::Crypto)?
-                .len()
-        } else {
-            let mut in_out = ciphertext.to_vec();
```

Comment:
@Liamsi I was a bit curious what case you're trying to handle here. It seems like it only helps in the event that the out buffer is up to the tag length smaller than the input ciphertext, but any shorter than that and the code below will still panic when it slices `out` out-of-bounds